omod/src/main/webapp/resources/scripts/htmlForm.js

```diff
@@ -258,24 +278,50 @@
     // TODO: these methods (getEncounter*Date*) will have to be modified when/if we switch datepickers
     // TODO: could/should be generalized so as not to be datepicker dependent?
 
-    htmlForm.setEncounterStartDateRange = function(date) {
+    //Set encounterDate datepicker start date
+    htmlForm.setEncounterStartDateRange = function(date , handletimezones) {
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'minDate', date);
+            if (handletimezones) {
+                var startDateClientTimezone = new Date(date)
+                getField('encounterDate.value').datepicker('option', 'minDate', startDateClientTimezone)
+            }else{
+                getField('encounterDate.value').datepicker('option', 'minDate',  extractDate(date));
+            }
         }
     };
 
-    htmlForm.setEncounterStopDateRange = function(date) {
+    //Set encounterDate datepicker stop date
+    htmlForm.setEncounterStopDateRange = function(date, handletimezones) {
+        //Set browser timezone
+        if(jq("#encounterDate").find(".hfe-timezone")){
+            jq("#encounterDate").find(".hfe-timezone").val(Intl.DateTimeFormat().resolvedOptions().timeZone)
+        }
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'maxDate', date);
+            if (handletimezones) {
+                var stopDateClientTimezone = new Date(date)
```

Comment:
What does stopDateClientTimezone mean? The stop date or the client time zone?

Reply:
That is the stop date with the client timezone. It transforms the stop date with server TZ to client TZ

Reply:
In that case, i find the name confusing.

Reply:
The same as the above if I change it to `stopDateWithClientTimezone ` ? Or do you have a better idea?

Reply:
Awesome!